### PR TITLE
[DOC] Add max_shard_size parameter to shrink API page

### PR DESCRIPTION
**Pull request by gaobinlong**
Signed-off-by: gaobinlong <gbinlong@amazon.com>

### Description
Relates to https://github.com/opensearch-project/documentation-website/issues/2044.
The main change of this PR is adding a new parameter named `max_shard_size` to shrink API page, the parameter was added to OpenSearch core in this PR: [5229](https://github.com/opensearch-project/OpenSearch/pull/5229), so we need to add the parameter to the documentation.

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/2044

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Naarcha-AWS — APPROVED** [review]

File: _api-reference/index-apis/shrink-index.md [hunk 1]
```diff
@@ -58,8 +58,4 @@ You can use the request body to configure some index settings for the target ind
...
-
-{% comment %}
-The new parameter to be called max_shard_size. 
-max_primary_shard_size | Bytes | Sets the maximum size of a primary shard in the target index. For example, if this field is set to 100 GB, and the source index's primary shards total to 300 GB, then the target index has 3 primary shards of 100 GB each.
-{% endcomment %}
+max_shard_size | Bytes | Sets the maximum size of a primary shard in the target index. `max_shard_size` calculates the target index's primary shard count and the primary shards' storage in the source index. The primary shard count of the target index is the lowest factor of the source index's primary shard count which satisfies this field's requirement. For example, if `max_shard_size` is set to `100 GB`, and the source index's primary shards total to `300 GB` and the count is `6`, then the target index has **3 primary shards of 100 GB each**. If the source index's primary shards total to `600GB` and count is `5`, then the target index will have **5 primary shards**. Because `max_shard_size` conflicts with `index.number_of_shards` in `settings` you can not set both of them at the same time.
```
Comment (kolchfa-aws) [inline _api-reference/index-apis/shrink-index.md:61]: @gaobinlong After looking at the code, I think we might want to break this parameter out in a separate section with more explanations and a non-trivial example. We might also want to mention the max and min for the number of primary shards in the target index. Do you mind if I take over this PR and try to do this, and you can review for technical accuracy?

Comment (gaobinlong) [reply _api-reference/index-apis/shrink-index.md:61]: @kolchfa-aws sure, you can do it, and feel free to ping me if you need some help.

Comment (kolchfa-aws) [reply _api-reference/index-apis/shrink-index.md:61]: @gaobinlong Could you please review the new PR for technical accuracy? #2519

Comment (gaobinlong) [reply _api-reference/index-apis/shrink-index.md:61]: Sure，I've added some comments for the new PR, you can take a look.

Comment (kolchfa-aws) [reply _api-reference/index-apis/shrink-index.md:61]: @gaobinlong Somehow I don't see any comments on #2519. I don't know if it's a problem on my end. Would you mind confirming that they are there?

**Comment by kolchfa-aws** [comment]
Closing this PR in favor of #2519

**Pull request closed without merging**